climada/hazard/test/test_tc_cc.py

```diff
@@ -26,40 +26,51 @@
 class TestKnutson(unittest.TestCase):
     """Test loading funcions from the TropCyclone class"""
 
-    def test_get_pass(self):
+    def test_get_knutson_scaling_pass(self):
         """Test get_knutson_criterion function."""
-        criterion = tc_cc.get_knutson_criterion()
-        self.assertTrue(len(criterion), 20)
-        for crit_val in criterion:
-            self.assertTrue('year' in crit_val)
-            self.assertTrue('change' in crit_val)
-            self.assertTrue('variable' in crit_val)
-        self.assertEqual(criterion[0]['variable'], "frequency")
-        self.assertEqual(criterion[0]['change'], 1)
-        self.assertEqual(criterion[4]['variable'], "intensity")
-        self.assertEqual(criterion[4]['change'], 1.045)
-        self.assertEqual(criterion[-10]['basin'], "SP")
-        self.assertEqual(criterion[-10]['variable'], "frequency")
-        self.assertEqual(criterion[-10]['change'], 1 - 0.583)
-
-    def test_scale_pass(self):
-        """Test calc_scale_knutson function."""
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2050, rcp_scenario=45),
-                               0.759630751756698)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2070, rcp_scenario=45),
-                               0.958978483788876)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2060, rcp_scenario=60),
-                               0.825572149523299)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2080, rcp_scenario=60),
-                               1.309882943406079)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2090, rcp_scenario=85),
-                               2.635069196605717)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2100, rcp_scenario=85),
-                               2.940055236533517)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2066, rcp_scenario=26),
-                               0.341930203294547)
-        self.assertAlmostEqual(tc_cc.calc_scale_knutson(ref_year=2078, rcp_scenario=26),
-                               0.312383928930456)
+        criterion = tc_cc.get_knutson_scaling_factor(
+                    percentile='5/10',
+                    baseline=(1950, 2018))
+        self.assertEqual(criterion.shape, (21, 4))
+
+        self.assertEqual(criterion.columns[0], '2.6')
+        self.assertEqual(criterion.columns[1], '4.5')
+        self.assertEqual(criterion.columns[2], '6.0')
+        self.assertEqual(criterion.columns[3], '8.5')
+
+        self.assertAlmostEqual(criterion.loc[2030, '2.6'], -16.13547, 4)
+        self.assertAlmostEqual(criterion.loc[2050, '4.5'], -25.19448, 4)
+        self.assertAlmostEqual(criterion.loc[2070, '6.0'], -31.06633, 4)
+        self.assertAlmostEqual(criterion.loc[2100, '8.5'], -58.98637, 4)
+
+    def test_get_gmst_pass(self):
```

Comment:
Could you please write this as a simple unit test (it must be plainly obvious when the answer is correct or not). Using more complex / real data is better for an integration test.

Reply:
New/updated tests should not follow outdated test logic. Anyway, all testing of `get_gmst_info` is somewhat pointless. The function takes no arguments and only returns structured data. You should instead test if the data is correctly applied in other functions calling `get_gmst_info`